### ISSUE TITLE
Getting stack output?

### ISSUE THREAD

**Issue by azarboon** [awslabs/serverless-application-model#538]
**Description:**
To perform automatic integration tests on my project, somehow I need to get the api. There is already the plugin [serverless-stack-output](https://github.com/sbstjn/serverless-stack-output) for Serverless framework that serves the purpose. But I'm wondering how can I achieve similar thing by AWS SAM after I deploy my application?

Meanwhile, If I can somehow get my api's base url as well as individual endpoints, then I'm able to connect them and and perform tests against them.

**Expected result:**

To get API Gateway endpoints + other invocation channels in a json file.

**Comment by brettstack** [comment]
You can use the Outputs section of CloudFormation, and query for that using the API

**Comment by azarboon** [comment]
Thank you. Yes, and I'm now looking at [backend example](https://github.com/awslabs/serverless-application-model/blob/master/examples/2016-10-31/api_backend/template.yaml) which does so.

But following that example's way, I should define separate "output" for each function. This can easily lead to a mess when there are many functions in the app. Is there any way to make it shorter syntax? 

Also, each api endpoint (e.g. in this case its "/resource") should be written and updated manually. Which can is annoying. Is there any way or any !REF that can get URL endpoint?

**Comment by brettstack** [comment]
Agreed. I think it would be beneficial if SAM created some Outputs for you

**Comment by jfuss** [comment]
Updating labels.
